parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/offsets/BitSetEncoder.java

```diff
@@ -77,7 +74,7 @@ public BitSetEncoder(long length, OffsetSimultaneousEncoder offsetSimultaneousEn
 
         this.originalLength = length;
 
-        // prep bit set buffer
+        // prep bit set buffer, range check above
         bitSet = new BitSet(Math.toIntExact(length));
```

Comment:
Bike shedding alert :) 
just generally thinking - would it be worth doing try / catch ArithmeticException instead of explicit check for length above? atm - you are relying on the separate check above - which is fine as such - but at same it pure sequencing reliance.